src/main/flight/mixer.c

```diff
@@ -269,83 +273,123 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
+    float signYaw = (getRcDeflection(FD_YAW) < 0 ? 1 : -1) * (mixerConfig()->yaw_motors_reversed ? 1 : -1);
+
+    float stickDeflectionLength = sqrtf(sq(stickDeflectionPitchAbs) + sq(stickDeflectionRollAbs));
+
+    if (stickDeflectionYawAbs > MAX(stickDeflectionPitchAbs, stickDeflectionRollAbs)) {
+        // If yaw is the dominant, disable pitch and roll
+        stickDeflectionLength = stickDeflectionYawAbs;
+        signRoll = 0;
+        signPitch = 0;
+    } else {
+        // If pitch/roll dominant, disable yaw
+        signYaw = 0;
+    }
+
+    const float cosPhi = (stickDeflectionLength > 0) ? (stickDeflectionPitchAbs + stickDeflectionRollAbs) / (sqrtf(2.0f) * stickDeflectionLength) : 0;
+    const float cosThreshold = sqrtf(3.0f)/2.0f; // cos(PI/6.0f)
+
+    if (cosPhi < cosThreshold) {
+        // Enforce either roll or pitch exclusively, if not on diagonal
+        if (stickDeflectionRollAbs > stickDeflectionPitchAbs) {
             signPitch = 0;
         } else {
-            // If pitch/roll dominant, disable yaw
-            signYaw = 0;
+            signRoll = 0;
         }
+    }
+
+    // Calculate crashflipPower from stick deflection with a reasonable amount of stick deadband
+    float crashflipPower = stickDeflectionLength > CRASHFLIP_STICK_MINF ? stickDeflectionLength : 0.0f;
+
+    // calculate flipPower attenuators
+    float crashflipRateAttenuator = 1.0f;
+    float crashflipAttitudeAttenuator = 1.0f;
+    const float crashflipRateLimit = mixerConfig()->crashflip_rate * 10.0f; // eg 35 = no power by 350 deg/s
+    const float smallAttChange = 0.5f; // attenuation factors less that this will be ignored
+    const float smallAttChangeInv = 1.0f / smallAttChange;
+
+    // disable both attenuators if the user's crashflip_rate is zero
+    if (crashflipRateLimit > 0) {
+#ifdef USE_ACC
+        // Calculate an attenuator based on change of attitude (requires Acc)
+        // with Acc, crashflipAttitudeAttenuator will be zero after approx 90 degree rotation, and
+        // motors will stop / not spin while attitude remains more than ~90 degrees from initial attitude
+        // without Acc, the user must manually center the stick, or exit crashflip mode, or disarm, to stop the motors
+        // re-initialisation of crashFlip mode by arm/disarm is required to reset the initial tilt angle
+        if (sensors(SENSOR_ACC)) {
+            const float tiltAngle = getCosTiltAngle();  // -1 if flat inverted, 0 when 90° (on edge), +1 when flat and upright
+            if (isFirstTiltAngleRead) {
+                tiltAngleAtStart = tiltAngle;
+                isFirstTiltAngleRead = false;
+            }
+            const float attitudeChange = fmaxf(1.0f - fabsf(tiltAngle - tiltAngleAtStart), 0.0f);
+            // no attenuation unless a significant attitude change has occurred
+            crashflipAttitudeAttenuator = attitudeChange > smallAttChange ? 1.0f : attitudeChange * smallAttChangeInv;
```

Comment:
I agree the idea that `attitudeChange` of 1.0 means no change, and 0 means sufficient change to stop motors, seems confusing.  I couldn't think of a better descriptor.
Using 1.0 -> 0.0 does keep the code simple, since the output is a multiplier and 1 means use motors, 0 means no motors.
I added a comment to clarify :-) 

Re: smallAttChangeInv - isn't it cheaper to multiply than divide?  And we need to do this in two places.

Reply:
maybe `attitudeGain` or `attitudeAttenuator` or multiplier of factor? When I read  `Change`, I expect that larger value means bigger change.

> Re: smallAttChangeInv - isn't it cheaper to multiply than divide? And we need to do this in two places.

I'd expect compiler to handle it automatically (it is local variable), so there will be no performance penalty here. Storing `sometingInv` mainly makes sense when it is global (or file-level static) variable. And even then, it makes sense only when `someting` is not used in code - storing both is not worth it (there will be negligible performance improvement and it is another variable to keep in sync, another line of code etc.)

Reply:
OK thanks will remove the `Inv`.

How about `attitudeVariation` instead of `attitudeChange`?

To me, linguistically, a 'change' can be good, or bad; or positive, or negative.  So to be honest, I was happy with 'change' here; I was thinking it just means that the attitude has changed from the previous attitude (and not much more than that).

I'll use `attitudeVariation` for now.